src/query-builder/InsertQueryBuilder.ts

```diff
@@ -1081,4 +1101,496 @@ export class InsertQueryBuilder<
+                    } else if (typeof value === "function") {
+                        expression += value()
+
+                        // just any other regular value
+                    } else {
+                        if (this.connection.driver.options.type === "mssql")
+                            value = (
+                                this.connection.driver as SqlServerDriver
+                            ).parametrizeValue(column, value)
+
+                        // we need to store array values in a special class to make sure parameter replacement will work correctly
+                        // if (value instanceof Array)
+                        //     value = new ArrayParameter(value);
+
+                        const paramName = this.createParameter(value)
+
+                        if (
+                            DriverUtils.isPostgresFamily(
+                                this.connection.driver,
+                            ) &&
+                            this.connection.driver.spatialTypes.indexOf(
+                                column.type,
+                            ) !== -1
+                        ) {
+                            if (column.srid != null) {
+                                expression += `ST_SetSRID(ST_GeomFromGeoJSON(${paramName}), ${column.srid})::${column.type}`
+                            } else {
+                                expression += `ST_GeomFromGeoJSON(${paramName})::${column.type}`
+                            }
+                        } else if (
+                            this.connection.driver.options.type === "mssql" &&
+                            this.connection.driver.spatialTypes.indexOf(
+                                column.type,
+                            ) !== -1
+                        ) {
+                            expression +=
+                                column.type +
+                                "::STGeomFromText(" +
+                                paramName +
+                                ", " +
+                                (column.srid || "0") +
+                                ")"
+                        } else {
+                            expression += paramName
+                        }
+                    }
+                    if (this.connection.driver.options.type !== "mssql")
+                        expression += ` AS ${this.escape(column.databaseName)}`
+
+                    if (columnIndex === columns.length - 1) {
+                        if (valueSetIndex === valueSets.length - 1) {
+                            if (
+                                this.connection.driver.options.type === "oracle"
+                            ) {
+                                expression += " FROM DUAL"
+                            } else if (
+                                this.connection.driver.options.type === "sap"
+                            ) {
+                                expression += " FROM dummy"
```

Comment:
Also an option:

```ts
if (["oracle", "sap"].includes(this.connection.driver.options.type)) {
    expression += " FROM " + this.connection.driver.dummyTableName
}
```

BTW, in SAP HANA, the dummy table is in the SYS schema. And you can create a table with this name in any other schema, so it's safer to explicitly add the schema.

This will return two rows instead of one:
```sql
CREATE TABLE DUMMY LIKE SYS.DUMMY WITH DATA;
INSERT INTO DUMMY SELECT * FROM SYS.DUMMY;
SELECT * FROM DUMMY;
```